--- src/FL2000/src/fl2000_surface.c 2021-04-18 09:35:45.961838990 +0200
+++ fl2000_surface.c    2021-04-18 11:07:22.875338507 +0200
@@ -106,13 +106,13 @@
    case SURFACE_TYPE_VIRTUAL_FRAGMENTED_VOLATILE:
    case SURFACE_TYPE_VIRTUAL_FRAGMENTED_PERSISTENT:
        while (surface->pages_pinned != nr_pages) {
-           down_read(&current->mm->mmap_sem);
+           mmap_read_lock(current->mm);
            pages_pinned = fl2000_get_user_pages(
                surface->user_buffer,
                nr_pages,
                pages,
                NULL);
-           up_read(&current->mm->mmap_sem);
+           mmap_read_unlock(current->mm);
            if (pages_pinned <= 0) {
                dbg_msg(TRACE_LEVEL_ERROR, DBG_PNP,
                    "get_user_pages fails with %d\n", pages_pinned);
@@ -126,7 +126,7 @@
        break;

    case SURFACE_TYPE_VIRTUAL_CONTIGUOUS:
-       down_read(&current->mm->mmap_sem);
+       mmap_read_lock(current->mm);
        /*
         * work-around the user memory which is mapped from driver,
         * but with VM_IO, VM_PFNMAP flags. This API assumes the mmaped user addr
@@ -141,7 +141,7 @@
            pages,
            NULL);
        vma->vm_flags = old_flags;
-       up_read(&current->mm->mmap_sem);
+       mmap_read_unlock(current->mm);
        if (pages_pinned <= 0) {
            dbg_msg(TRACE_LEVEL_ERROR, DBG_PNP,
                "get_user_pages fails with %d\n", pages_pinned);
@@ -228,8 +228,7 @@
        surface->mapped_buffer = vm_map_ram(
            surface->pages,
            surface->nr_pages,
-           -1,
-           PAGE_KERNEL);
+           -1);
        if (surface->mapped_buffer == NULL) {
            dbg_msg(TRACE_LEVEL_ERROR, DBG_PNP, "vm_map_ram failed?");
            ret_val = -ENOMEM;
